src/process/conditions/index.ts

```diff
@@ -102,17 +93,14 @@ export const conditionalProcess = (context: ConditionsContext, isHidden: Conditi
         scope.conditionals.push(conditionalComp);
     }
 
-    if (skipOnServer(context)) {
-        return false;
-    }
-
     conditionalComp.conditionallyHidden = conditionalComp.conditionallyHidden || isHidden(context);
     if (conditionalComp.conditionallyHidden) {
         const info = componentInfo(component);
         if (info.hasColumns || info.hasComps || info.hasRows) {
             // If this is a container component, we need to add all the child components as conditionally hidden as well.
             Utils.eachComponentData([component], row, (comp: Component, data: any, compRow: any, compPath: string) => {
                 if (comp !== component) {
+                    // the path set here is not the absolute path, but the path relative to the parent component
                     scope.conditionals?.push({ path: getComponentPath(comp, compPath), conditionallyHidden: true });
```

Comment:
this sets a partial path to the parent `<parent-key>.<field-key>` not an absolute path